tests/test_generate.py

```diff
@@ -66,21 +68,49 @@ def _test_greedy_decoding(model_name, device="cpu", use_cache=False, decoder_onl
     return np.array(results)
 
 
-testdata = [
+greedy_testdata = [
     ("t5-small", True, False),
     ("t5-small", False, False),
 ]
 
+beam_search_testdata = [
+    ("facebook/bart-base", False, False)
+]
 
 @is_trainium_test
-@pytest.mark.parametrize("model_name, use_cache, decoder_only", testdata)
+@pytest.mark.parametrize("model_name, use_cache, decoder_only", greedy_testdata)
 def test_greedy_decoding(model_name, use_cache, decoder_only):
     os.environ["XLA_USE_BF16"] = "0"
-    xla_neuron_samples_fp32 = _test_greedy_decoding(model_name=model_name, device="xla", decoder_only=decoder_only)
+    xla_neuron_samples_fp32 = _test_generative_decoding(model_name=model_name, device="xla", decoder_only=decoder_only)
     os.environ["XLA_USE_BF16"] = "1"
-    xla_neuron_samples_bf16 = _test_greedy_decoding(model_name=model_name, device="xla", decoder_only=decoder_only)
+    xla_neuron_samples_bf16 = _test_generative_decoding(model_name=model_name, device="xla", decoder_only=decoder_only)
+
+    cpu_samples = _test_generative_decoding(model_name=model_name, device="cpu", decoder_only=decoder_only)
+
+    assert np.array_equal(cpu_samples, xla_neuron_samples_fp32), "XLA Neuron FP32 output doesn't match CPU only output"
+    assert np.array_equal(cpu_samples, xla_neuron_samples_bf16), "XLA Neuron bf16 output doesn't match CPU only output"
 
-    cpu_samples = _test_greedy_decoding(model_name=model_name, device="cpu", decoder_only=decoder_only)
+@is_trainium_test
+@pytest.mark.parametrize("model_name, use_cache, decoder_only", beam_search_testdata)
+def test_beam_search_decoding(model_name, use_cache, decoder_only):
+    os.environ['NEURON_CC_FLAGS'] = '--model-type=transformer --enable-saturate-infinity'
```

Comment:
If we need that, we should add this to the generate method no?

Reply:
It is actually specific to the BART model.